code/modules/clothing/clothing.dm

```diff
@@ -217,6 +218,22 @@
 		for(var/A in armor)
 			armor[A] -= rand(armor[A]/3, armor[A])
 
+/obj/item/clothing/attack(var/mob/living/M, var/mob/living/user, def_zone, var/originator = null)
+	if(iscarbon(user))
```

Comment:
This code is indented really badly, which makes reading it pretty painful.

Instead of doing this:
```dm
if (x)
    if (y)
        // Thing

..()
```
Change the style to be more like this:
```dm
if (!x || !y)
    ..()
    return

// Thing
```

Reply:
I had to take a moment to stare at the suggestion until I understood it. For a moment I had a worry that if X failed it'd try Y only to realize that X failing would cause a true condition, thus executing the 'if' block right away since it doesn't have to check if the Y failed to cause a true, since, of course, it'd be pointless to check in a 'OR' statement if the first time was a success.

I'll have to keep that in mind in my other projects to prevent the dreaded pyramid hell.

Reply:
I recommend collapsing the `if` statements, but not how he said to
```
if(x && y)
    //Thing
```

Reply:
>This code is indented really badly

It's technically not an indentation problem, it's just several ifs in succession.
Those ifs can't be checked separately.
You need to know that's it's a carbon to check the intent. Intent is not defined for mobs.
You need to know that `M` is a human to do the polymorphism trick, and you need to do that to check if it's a plasmaman. `isplasmaman()` doesn't work with mobs because `species` is not defined.

The only control that could be collapsed is the one about the clothing.

Reply:
Just so you know, writing `(x && y)` wouldn't evaluate x and y "concurrently". You're guaranteed x will be evaluated before y.

Reply:
Oh. Well I did _not_ know that.
Guess it's time to do another commit then :^)

Reply:
You can still check for both things in one `if` statement. Eg. if X needs to be true before Y can be executed
````dm
if(X && Y)
    //Thing
````
If `X` returns `False` it'd skip executing `//Thing` since in a `AND` condition it requires everything to be `True`, so checking if `Y` is true would be moot, thus [Short-circuit evaluation](https://en.wikipedia.org/wiki/Short-circuit_evaluation). So if `X` fails, it won't bother trying to execute `Y`.

Same logic applies to the suggested if indents are to be discouraged.
````dm
if(!X || !Y)
    ..()
   return
//Thing
````
Assuming `X` returns `False`, it becomes `True` due to the `!` operator, causing Short-circuit evaluation to prevent `Y` from executing.

Also bothered @PJB3005 to check about having the evaluations happen on multiple lines, since moving them into one line would cause readability issues, and it's done with a backslash if you want to move to the next line without causing the compiler to throw a fit.
````dm
if(   CheckThis()  \
   && AndThis()    \
   && AlsoThisToo())
   //Thing
````

Reply:
Thanks but I know how to code, I just thought that if he saw `if (x && y)` he would check x, then check y, and only then check x && y. Which would make no sense in this case since `y` was not properly defined.
Yeah I'm going to do what PJB suggested, though since it reworks the code in an arcane way I'm gonna test it again.
Though this is a matter of taste, I've coded mostly with ifs in succession so to me it's natural. It also do the same amount of calculation.